### ISSUE TITLE
Scan More than 200 books on Goodreads

### ISSUE THREAD

**Issue by driscoll42** [isaactbock/library-scan#13]
Opening a separate ticket for this for future reference.

**Is your feature request related to a problem? Please describe.**
Currently Library Scan only scans the most recent 200 books on your Goodreads To-Read and my To Read list is ~403 long and I expect will grow over time. 

**Describe the solution you'd like**
Ideally Library Scan would scan my entire To Read list for books on Overdrive. Looking at the Goodreads URL, there's a "&page=2" that could be added to the to read list URL and then in the queryGoodreads function, some kind of while loop could be added that checks if there's books or not and if not breaks should work. Or if there's a way to determine the total length of the To Read list a for loop would be safer I imagine.

**Describe alternatives you've considered**
Currently I could resort things on my To Read list, but that would be something of a hit or miss game and would require quite a bit of manual work.

**Comment by driscoll42** [comment]
I tried just wrapping the entire fetch block in a for loop

`	for (let page_num = 1; page_num < 3; page_num++) {
`

Which obviously doesn't scale, but it succeeded in checking everything on my to read list. There *are* issues in that the progress bar clearly doesn't make sense (bounces between each batch of 200 so looks like 128/200 then 140/200 then 110/200 every second), but it *works* insomuch as it is scanning my entire library though not production ready (most obvious issue is that I'm hardcoding a max page_num).

**Comment by driscoll42** [comment]
Playing with this some more, unfortunately I'm not a javascript dev, but here's the findings so far:

Having a loop such as:

	for (let page_num = 1; page_num < 3; page_num++) { 

works to pull in more than 200 books just fine. However if page_num is greater than the number that would actually return a real page, it'll display an error, but only briefly before showing the progress bar. Then once the progress bar is displayed, it bounces around a fair bit, it seems like javascript executes greedily perhaps?


I expect the steps that need to be done are:
1. Add a while loop with everything from fetch("https://..... through the catch block to keep trying until an exception is returned for an invalid page in which case it breaks. However this catch would need to be somehow different than the generic catch already being caught
2. Store all the results in bookArray and ToRead but don't start calling queryOverdrive until the while loop is complete
3. Call queryOverdrive normally to start querying for books

**Comment by driscoll42** [comment]
Looking at the XML response from Goodreads, it starts with, for mine for example:

<reviews start="1" end="100" total="785">

So instead of a while loop, it'd be pretty easy to grab that, and know there's a need for only four loops with 200, should remove the need for some weird catch error handling.
